helpers/html-elem-context-provider.js

```diff
@@ -0,0 +1,32 @@
+const attributeCallbacks = new Map();
+
+const contextObserver = new MutationObserver(mutations => {
+	mutations.forEach(mutation => {
+		const callbacks = attributeCallbacks.get(mutation.attributeName);
+		if (callbacks !== undefined && callbacks.length > 0) {
+			callbacks.forEach(callback => callback());
+		}
+	});
+});
+
+let observerInitialized = false;
+
+export function getContext(attributeName) {
```

Comment:
As I mentioned in the BSI PR, I think whether we even want this in `core` depends on whether we think there's value in other consumers being able to import this to set up their own fake hosts using the HTML element. I think it might more sense to have the `lms-context-provider` provide a `test-host` export that can do this without relying on the document, however, in which case this stuff should probably just live in (and be managed by) BSI. 🤷

Reply:
I'm personally leaning towards trying to find a nicer "automatic" solution for tests & demos. This doesn't feel very core-y... like _maybe_ if it were made more generic as a way to observe attributes on an arbitrary element but even that feels very specific to this case.

Reply:
Yeah, I think I agree. I'm inclined to move this logic into BSI (we need this behaviour one way or the other in the LMS) and expose something like that `test-host` in the provider repo itself. `core` stuff like the `html-block` can keep using the attribute controller as it currently does, and if that's ever something we want to change, we can cross that bridge as we come to it.